app/services/notifications/moderation/send.rb

```diff
@@ -0,0 +1,39 @@
+# Send notifications from moderation
+module Notifications
+  module Moderation
+    class Send
+      def initialize(available_moderators, notifiable)
+        @available_moderators = available_moderators
+        @notifiable = notifiable
+      end
+
+      def self.call(*args)
+        new(*args).call
+      end
+
+      delegate :user_data, :comment_data, to: Notifications
+
+      def call
+        # notifiable is currently only comment
+        moderator = available_moderators.sample
```

Comment:
this loads in memory all the moderators, just to choose one at random. I'd let SQL do it:

```ruby
available_moderators.order(Arel.sql("RANDOM()")).first
```

The code is also using `available_moderators` only here, so maybe the service object can just receive a moderator and the notifiable, like:

```ruby
def initialize(moderator, notifiable)
```

Reply:
Probably, in this case, it can be just `available_moderators.order("RANDOM()")` ?

Reply:
I removed the .sample and added Arel.sql( with just .order, there are warnings that it will be mandatory in the next version of Rails).